commandapi-core/src/main/java/dev/jorel/commandapi/arguments/ArgumentParseExceptionContext.java

```diff
@@ -0,0 +1,14 @@
+package dev.jorel.commandapi.arguments;
+
+import dev.jorel.commandapi.exceptions.WrapperCommandSyntaxException;
+
+/**
+ * A record containing information on why an Argument failed to parse.
+ *
+ * @param exception The CommandSyntaxException that was thrown when the Argument failed to parse
+ * @param sender The CommandSender who sent the command that caused the exception
+ * @param input The raw object returned by the initial Brigadier parse for the Argument
+ * @param previousArguments The previously parsed arguments that came before this argument
+ */
+public record ArgumentParseExceptionContext<CommandSender>(WrapperCommandSyntaxException exception, CommandSender sender, Object input, Object[] previousArguments) {
```

Comment:
> `Object[] previousArguments` may no longer be the right thing to use here, given we have the new `CommandArguments` class. Would that be a better fit here?

Other developer-facing records like `CustomArgumentInfo` and `SuggestionInfo` still use `Object[]` in reference to `previousArgs`. I think we should switch these to use `CommandArguments` on dev/dev first to avoid doing the same work twice, since many of the argument-parsing related methods need to be switched to use and return `CommandArgument` to make these work.

Reply:
> Needs JavaDocs for the various fields.

Is that how record java docs work? The way the javadocs currently are seem to work for me:
<img width="1365" alt="Screen Shot 2023-02-01 at 6 02 47 PM" src="https://user-images.githubusercontent.com/46540330/216186033-45e95222-60b2-4f88-bcf8-1441da36f7d5.png">
<img width="834" alt="Screen Shot 2023-02-01 at 6 04 40 PM" src="https://user-images.githubusercontent.com/46540330/216186168-141564f7-5a17-4aa2-b5ad-4f68f3d8a743.png">

Reply:
> Is that how record java docs work?
No clue. Apparently nobody really knows how record JavaDocs are "supposed" to work because it was never specified in the JEP when records were added. In Eclipse, JavaDocs for record methods don't exist (and neither do they exist for record value field access e.g. `blah.exception` as opposed to `myobject.exception()`) unless the JavaDoc comment is present on the parameter itself.

I've been doing this for all of the other records in the CommandAPI. I don't know if this is a standard or not, but it seems to do the job and I don't mind duplicating JavaDoc comments (i.e. having JavaDocs both above the constructor and next to parameters) - the key thing is that the JavaDocs exist when viewed.

Doxygen doesn't seem to support parsing JavaDocs on parameters for records, but then again Doxygen treats records as functions, not classes.